src/main/java/reposense/model/RepoLocation.java

```diff
@@ -61,11 +68,27 @@ public RepoLocation(String location) throws InvalidLocationException {
         }
 
         this.location = location;
+        this.isLocal = isLocalRepo(location);
         String[] repoNameAndOrg;
         if (location.isEmpty()) {
             repoNameAndOrg = new String[] {"", "", UNSUPPORTED_DOMAIN_NAME};
-        } else if (isLocalRepo(location)) {
-            repoNameAndOrg = getLocalRepoNameAndOrg(location);
+        } else if (this.isLocal) {
+            String[] localRepoNameAndOrg = getLocalRepoNameAndOrg(location);
+            this.localRepoName = localRepoNameAndOrg[0];
+            this.localOrganization = localRepoNameAndOrg[1];
+
+            Map<String, String> remotes = GitRemote.getRemotes(location);
+            String newLocation = remotes.size() == 0
+                    ? location
+                    : remotes.containsKey("origin(fetch)")
+                    // Get fetch remote named 'origin' if possible
+                    ? remotes.get("origin(fetch)")
```

Comment:
I think for lines 83 and 85, the constant `origin(fetch)` should be stored in the `GitRemote` class. Either that or create some kind of static accessor method in `GitRemote` that uses this key instead.

Reply:
That's a good idea! Thanks for the suggestion.